prometheus_hardware_exporter/collector.py

```diff
@@ -109,3 +112,261 @@ def fetch(self) -> List[Payload]:
     def process(self, payloads: List[Payload], datastore: Dict[str, Payload]) -> List[Payload]:
         """Process the payload if needed."""
         return payloads
+
+
+class IpmiDcmiCollector(BlockingCollector):
+    """Collector for ipmi dcmi metrics."""
+
+    ipmi_dcmi = IpmiDcmi()
+
+    @property
+    def specifications(self) -> List[Specification]:
+        """Define dcmi metric specs."""
+        return [
+            Specification(
+                name="ipmi_dcmi_power_cosumption_watts",
+                documentation="Current power consumption in watts",
+                labels=[],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ipmi_dcmi_command_success",
+                documentation="Indicates if the ipmi dcmi command is successful or not",
+                labels=[],
+                metric_class=GaugeMetricFamily,
+            ),
+        ]
+
+    def fetch(self) -> List[Payload]:
+        """Load current power from ipmi dcmi power statistics."""
+        current_power_payload = self.ipmi_dcmi.get_current_power()
+
+        if not current_power_payload:
+            logger.error("Failed to fetch current power from ipmi dcmi")
+            return [Payload(name="ipmi_dcmi_command_success", labels=[], value=0.0)]
+
+        payloads = [
+            Payload(
+                name="ipmi_dcmi_power_cosumption_watts",
+                labels=[],
+                value=current_power_payload["current_power"],
+            ),
+            Payload(name="ipmi_dcmi_command_success", labels=[], value=1.0),
+        ]
+        return payloads
+
+    def process(self, payloads: List[Payload], datastore: Dict[str, Payload]) -> List[Payload]:
+        """Process the payload if needed."""
+        return payloads
+
+
+class IpmiSensorsCollector(BlockingCollector):
+    """Collector for ipmi sensors data."""
+
+    ipmimonitoring = IpmiMonitoring()
+
+    @property
+    def specifications(self) -> List[Specification]:
+        """Define ipmi sensors metrics specs."""
+        return [
+            Specification(
+                name="ipmi_temperature_celsius",
+                documentation="Temperature measure from temperature sensors",
+                labels=["id", "name", "state", "unit", "event"],
```

Comment:
What is "event"? Will it contains many different value? 
If it is, I will suggest to remove it to reduce the dimension. 
Similar situation to other metrics. 
What do you think? 
We can also host a small discussion to discuss how to handle "event" if it is still a mandatory information for the metrics.